lib/ews/soap/builders/ews_builder.rb

```diff
@@ -1205,6 +1205,54 @@ def user_configuration_properties!(cfg_prop)
       @nbuild[NS_EWS_MESSAGES].UserConfigurationProperties(cfg_prop)
     end
 
+    # Build the SearchParameters element.
+    #  Example parameter `opts`:
+    #  {
+    #    traversal: "Deep",
+    #    t_restriction: (see `t_restriction!`),
+    #    base_folder_ids: (see `base_folder_ids!`)
+    #  }
+    def search_parameters!(opts)
+      mode = opts.delete(:traversal)
+
+      @nbuild[NS_EWS_TYPES].SearchParameters(Traversal: mode) do
+        opts.each_pair do |k, v|
+          ftype = "#{k}!".to_sym
+          if self.respond_to?(ftype)
+            self.send ftype, v
+          else
+            raise Viewpoint::EWS::EwsNotImplemented, "#{ftype} not implemented as a builder."
+          end
+        end
+      end
+    end
+
+    # Build the BaseFolderIds element.
+    #  Example parameter `folder_ids`:
+    #  [{ id: :msgfolderroot, act_as: 'someone@else.com'}]
+    def base_folder_ids!(folder_ids)
+      @nbuild[NS_EWS_TYPES].BaseFolderIds do
+        folder_ids.each { |fid| dispatch_folder_id!(fid) }
+      end
+    end
+
+    # Build the (type) Restriction element. Note that although it looks similar
+    #  to the other Restriction in (message) namespace, they are not interchangable!
+    #  Example parameter `restriction`:
+    #  {
+    #    is_greater_than: [
+    #      { field_uRI: { field_uRI: "item:DateTimeCreated"} },
```

Comment:
`uRI`?  Really?

I presume this is Microsoft's fault.

Reply:
kinda, it's ViewPoint translating between Ruby snake case and MS semi-camel case (FieldURI)